File: _scratch/run/simplify_path/sol.py
class Solution:
    def simplifyPath(self, s):
        st = []
        for part in s.split("/"):
            if part == "" or part == ".":
                continue
            if part == "..":
                if st: st.pop()
            else:
                st.append(part)
        return "/" + "/".join(st)
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append(str(Solution().simplifyPath(_s)))
print('\n'.join(_o))

File: _scratch/run/simplify_path/sol.js
/**
 * @param {string} s
 * @return {string}
 */
var simplifyPath = function(s) {
    const st = [];
    for (const part of s.split('/')) {
        if (part === '' || part === '.') continue;
        if (part === '..') { if (st.length) st.pop(); }
        else st.push(part);
    }
    return '/' + st.join('/');
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(String(simplifyPath(_s)));}
console.log(_o.join('\n'));